Visual Studio Project Template C#/PluginInfrastructure/GatewayDomain.cs

```diff
@@ -63,14 +63,19 @@ public int Value
     /// </summary>
     public class Position : IEquatable<Position>
     {
-        private readonly int pos;
+        private readonly long pos;
 
-        public Position(int pos)
+        public Position(long pos)
         {
             this.pos = pos;
         }
 
-        public int Value
+        public Position(IntPtr pos)
+        {
+            this.pos = IntPtr.Size == sizeof(Int32)? pos.ToInt32(): pos.ToInt64();
```

Comment:
could have been simply as follows too, since we are anyway casting it to long implicitly:

 this.pos = pos.ToInt64();